dpipe/predict/shape.py

```diff
@@ -79,10 +82,45 @@ def wrapper(x, *args, **kwargs):
     return decorator
 
 
+class FinishedToken:
+    pass
+
+
+class AsyncPmap:
```

Comment:
Seems like its quite general class. Mind to move it to `dpipe.itertools` with more task-agnostic namings. e.g. `__result_queue` instead of `__prediction_queue`

Reply:
done